src/RoutingData.hs

```diff
@@ -62,3 +66,11 @@ splitBucket kb splitIndex
      where (zeros, ones) = L.partition (\(nid, _) -> (nid!!splitIndex) == '0') kb
            newIndex = splitIndex + 1
 
+nodeDistance :: ID -> ID -> Int
+nodeDistance x y | x == y    = 0
+                 | otherwise = bitsToDec $ toBits $ zipWith charXor x y
+             where charXor = (xor) `on` digitToInt
+                   toBits = map intToDigit
+
+bitsToDec :: ID -> Int
+bitsToDec = foldl' (\acc x -> acc * 2 + digitToInt x) 0
```

Comment:
I really like this. Foldl folds from the left, so you can think of

`foldl' (\acc x -> acc * 2 + digitToInt x) 0 "1101"`
as
`((((0 * 2 + 1) * 2 + 1) * 2 + 0) * 2 + 1)`
As you add more digits, the digits you've walked get multiplied by another 2. mmmmmmmmmm recursion